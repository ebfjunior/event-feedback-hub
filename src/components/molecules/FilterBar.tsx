import type { SortOption, EventListItem } from '@/types/api';
import { EventSelect } from '@/components/molecules/controls/EventSelect';
import { RatingSelect } from '@/components/molecules/controls/RatingSelect';
import { SortToggle } from '@/components/molecules/controls/SortToggle';

export type FilterBarProps = {
  events: EventListItem[];
  fixedEventId?: string;
  eventId?: string;
  rating?: number;
  sort: SortOption;
  onEventChange?: (eventId: string | undefined) => void;
  onRatingChange?: (rating: number | undefined) => void;
  onSortChange: (sort: SortOption) => void;
};

export function FilterBar({ events, fixedEventId, eventId, rating, sort, onEventChange, onRatingChange, onSortChange }: FilterBarProps) {
  return (
    <div className="flex flex-wrap items-center gap-2">
      {!fixedEventId && <EventSelect events={events} value={eventId} onChange={onEventChange} />}
      <RatingSelect value={rating} onChange={onRatingChange} />
      <SortToggle value={sort} onChange={onSortChange} />
    </div>
  );
}


