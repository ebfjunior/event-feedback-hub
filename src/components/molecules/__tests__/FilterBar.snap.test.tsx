import React from 'react';
import { render } from '@testing-library/react';
import { FilterBar } from '../FilterBar';

vi.mock('../controls/EventSelect', () => ({
	EventSelect: (props: any) => <div data-mock="EventSelect" {...props} />,
}));
vi.mock('../controls/RatingSelect', () => ({
	RatingSelect: (props: any) => <div data-mock="RatingSelect" {...props} />,
}));
vi.mock('../controls/SortToggle', () => ({
	SortToggle: (props: any) => <div data-mock="SortToggle" {...props} />,
}));

describe('FilterBar (snapshot)', () => {
	it('renders with all controls', () => {
		const { container } = render(
			<FilterBar
				events={[{ id: 'e1', name: 'Event 1' }]}
				sort="newest"
				eventId={undefined}
				rating={undefined}
				onSortChange={() => {}}
			/>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('hides EventSelect when fixedEventId present', () => {
		const { container } = render(
			<FilterBar
				events={[{ id: 'e1', name: 'Event 1' }]}
				sort="newest"
				fixedEventId="e1"
				onSortChange={() => {}}
			/>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});


