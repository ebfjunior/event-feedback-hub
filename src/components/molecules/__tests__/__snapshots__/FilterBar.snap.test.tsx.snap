// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterBar (snapshot) > hides EventSelect when fixedEventId present 1`] = `
<div
  class="flex flex-wrap items-center gap-2"
>
  <div
    data-mock="RatingSelect"
  />
  <div
    data-mock="SortToggle"
    value="newest"
  />
</div>
`;

exports[`FilterBar (snapshot) > renders with all controls 1`] = `
<div
  class="flex flex-wrap items-center gap-2"
>
  <div
    data-mock="EventSelect"
    events="[object Object]"
  />
  <div
    data-mock="RatingSelect"
  />
  <div
    data-mock="SortToggle"
    value="newest"
  />
</div>
`;
